sidecar.go

```diff
@@ -64,8 +64,15 @@ const (
 	TelegrafSecretLabelPod        = "telegraf.influxdata.com/pod"
 )
 
+// sidecarHandlerInterface defines minimum interface for sidecar handler part related to
+// configurations which also allows test code to provide different implementations.
+type sidecarHandlerInterface interface {
```

Comment:
Go style: avoid suffixing a type name with the type's kind, i.e. prefer `sidecarHandler` to sidecarHandlerInterface`

If it clashes with a concrete impl, prefer renaming the concrete impl, e.g. `sidecarHandlerImpl`

Reply:
I only need one function - `assembleConf` - so I decided to change this so that I don't have to create the interface to also implement a mock of it.

For `classDataHandler` where I also defined an interface for it, I used `classDataHandler` for the interface and `directoryClassDataHandler` for implementation that reads a directory (volume).